### PR TITLE
Introduce Travis CI

### PR DESCRIPTION
**Pull request by mseri**
There are actually two failures due to rounding errors that are not invalidating the tests. This will need further investigation but I think is out of scope for this PR.

```
ocaml setup.ml -test
....................F............F
==============================================================================
Failure: qmp:1:parse_print:16:block-io-error.json
expected: {"event":"BLOCK_IO_ERROR","timestamp":{"seconds":1265044230,"microseconds":450485}}
but got: {"event":"BLOCK_IO_ERROR","timestamp":{"seconds":1265044230,"microseconds":450484}}
------------------------------------------------------------------------------
==============================================================================
Failure: qmp:1:parse_print:3:powerdown.json
expected: {"event":"POWERDOWN","timestamp":{"seconds":1258551470,"microseconds":802383}}
but got: {"event":"POWERDOWN","timestamp":{"seconds":1258551470,"microseconds":802382}}
------------------------------------------------------------------------------
Ran: 34 tests in: 0.01 seconds.
FAILED: Cases: 34 Tried: 34 Errors: 0 Failures: 2 Skip:  0 Todo: 0 Timeouts: 0.
```

### CODE REVIEW

File: opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
+opam-version: "1.2"
```
Comment (gaborigloi) [inline opam]: Is this `opam` file deliberately different from the one in `xs-opam`?:
```
> colordiff -u opam ../xs-opam/packages/xs/qmp.0.9.4/opam
--- opam        2017-06-19 14:09:54.326851950 +0100
+++ ../xs-opam/packages/xs/qmp.0.9.4/opam       2017-06-16 17:52:39.525686498 +0100
@@ -11,11 +11,6 @@
   ["./configure" "--prefix" prefix]
   [make]
 ]
-build-test: [
-    ["./configure" "--prefix" prefix "--enable-tests"]
-    [make "test"]
-]
-install: [make "install" "BINDIR=%{bin}%"]
 remove: [
   [make "uninstall" "BINDIR=%{bin}%"]
   ["ocamlfind" "remove" "qmp"]
@@ -28,3 +23,4 @@
   "cmdliner"
 ]
 dev-repo: "git://github.com/xapi-project/ocaml-qmp"
+install: [make "install" "BINDIR=%{bin}%"]
```

Comment (mseri) [inline opam]: Yes, I added the `build-test` that was missing from `xs-opam` and moved the `install` stanza in a more appropriate place

Comment (gaborigloi) [inline opam]: I see, this `opam` file is actually better, we should sync them at some point.

File: opam [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+install: [make "install" "BINDIR=%{bin}%"]
+remove: [
+  [make "uninstall" "BINDIR=%{bin}%"]
+  ["ocamlfind" "remove" "qmp"]
+]
+depends: [
+  "oasis"     {build}
+  "ocamlfind" {build}
+  "yojson"
+  "ounit"
```
Comment (gaborigloi) [inline opam]: Could we have this as a `{test}` dependency? I think that would work, as the only module using OUnit is only enabled for `build-test`.

File: opam (hunk 1 above)
Comment (mseri) [inline opam]: Sure

**Comment by gaborigloi** [comment]
We need to port it to `OUnit2`, I tested it with OUnit2 and it actually makes the test run fail if one of the tests fails, but apparently it's not the case for OUnit1.

File: opam [hunk 3]
```diff
@@ -0,0 +1,30 @@
+opam-version: "1.2"
+maintainer: "dave.scott@eu.citrix.com"
```
Comment (gaborigloi) [inline opam]: I think we should also update this field to `maintainer: "xen-api@lists.xen.org"`.

File: opam [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+build-test: [
+    ["./configure" "--prefix" prefix "--enable-tests"]
+    [make "test"]
+]
+install: [make "install" "BINDIR=%{bin}%"]
+remove: [
+  [make "uninstall" "BINDIR=%{bin}%"]
+  ["ocamlfind" "remove" "qmp"]
+]
+depends: [
```
Comment (gaborigloi) [inline opam]: Should we also add the `base-unix` package, as the `unix` ocamlfind package is a dependency in the `_oasis` file?

**Pull request merged**